extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java

```diff
@@ -144,6 +144,7 @@ public UserRoleDaoResource( final IRoleAuthorizationPolicyRoleBindingDao roleBin
     } )
   public Response createUser( User user ) {
     try {
+      user.setPassword( b64DecodePassword( user.getPassword() ) );
```

Comment:
can't the user object received here already have the encoded password? 
if not, what i'm not getting is, if the we are transmitting the password in clear text, why are we decoding the password here? shouldn't we be encoding the password instead?

Reply:
On file UserRolesAdminPanelController.java at line 95 it is create a json object with password encoded and then send the request, that request came to here, so it is necessary decode from base 64 to plain string. 
Later userRoleDaoService.createUser( user ) will create and user in database with a non-reversible encoded password